src/support/hash_city.c

```diff
@@ -86,33 +86,60 @@ static uint32_t UNALIGNED_LOAD32(const char *p) {
 	return (result);
 }
 
-#if !defined(WORDS_BIGENDIAN)
+#ifdef _MSC_VER
 
-#define	uint32_in_expected_order(x) (x)
-#define	uint64_in_expected_order(x) (x)
+#include <stdlib.h>
+#define	bswap_32(x) _byteswap_ulong(x)
+#define	bswap_64(x) _byteswap_uint64(x)
 
-#else
+#elif defined(__APPLE__)
 
-#ifdef __APPLE__
-/* Mac OS X / Darwin features */
+// Mac OS X / Darwin features
 #include <libkern/OSByteOrder.h>
 #define	bswap_32(x) OSSwapInt32(x)
 #define	bswap_64(x) OSSwapInt64(x)
 
-#elif defined(__sun)
+#elif defined(__sun) || defined(sun)
 
 #include <sys/byteorder.h>
-#define	bswap_32 BSWAP_32
-#define	bswap_64 BSWAP_64
+#define	bswap_32(x) BSWAP_32(x)
```

Comment:
I'm surprised about these changes - I'd thought this would get very simple with ``endian.h``. What am I missing?